From c84d3a04d5700b1c8416ae275d7c9e2e60b4c13c Mon Sep 17 00:00:00 2001
From: Tiberiu Breana <andrei-tiberiu.breana@nxp.com>
Date: Thu, 6 Jul 2017 18:28:11 +0300
Subject: [PATCH 2548/5242] MLK-15141-1: PCI: imx: Add epdev_on regulator for
 8QM WiFi

commit  19d8c6fe5038a35e89a9ec29274517b4e4c616b0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add the epdev_on regulator to power up the WiFi module
on the iMX8QM board.
This regulator needs to be powered up before the pcie
link, in order for the WiFi module to work.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Tiberiu Breana <andrei-tiberiu.breana@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index 7b3e53fe..31924c6 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -89,6 +89,7 @@ struct imx6_pcie {
 	void __iomem		*phy_base;
 	struct regulator	*pcie_phy_regulator;
 	struct regulator	*pcie_bus_regulator;
+	struct regulator	*epdev_on;
 };
 
 /* Parameters for the waiting for PCIe PHY PLL to lock on i.MX7 */
@@ -1969,6 +1970,15 @@ static int __init imx6_pcie_probe(struct platform_device *pdev)
 				"pcie clock source missing or invalid\n");
 			return PTR_ERR(imx6_pcie->pcie_inbound_axi);
 		}
+
+		imx6_pcie->epdev_on = devm_regulator_get(&pdev->dev,
+							 "epdev_on");
+		if (IS_ERR(imx6_pcie->epdev_on))
+			return -EPROBE_DEFER;
+
+		ret = regulator_enable(imx6_pcie->epdev_on);
+		if (ret)
+			dev_err(dev, "failed to enable the epdev_on regulator\n");
 	} else {
 		imx6_pcie->iomuxc_gpr =
 		 syscon_regmap_lookup_by_compatible("fsl,imx6q-iomuxc-gpr");
-- 
1.7.9.5

